src/v0/destinations/braze/util.js

```diff
@@ -299,9 +306,77 @@ const processDeduplication = (userStore, payload) => {
   return null;
 };
 
+const processBatch = (transformedEvents) => {
+  const { destination } = transformedEvents[0];
+  const attributesArray = [];
+  const eventsArray = [];
+  const purchaseArray = [];
+  const successMetadata = [];
+  const failureResponses = [];
+  for (const transformedEvent of transformedEvents) {
+    if (transformedEvent?.batchedRequest?.body?.JSON) {
+      const { attributes, events, purchases } = transformedEvent.batchedRequest.body.JSON;
+      if (Array.isArray(attributes)) {
+        attributesArray.push(...attributes);
+      }
+      if (Array.isArray(events)) {
+        eventsArray.push(...events);
+      }
+      if (Array.isArray(purchases)) {
+        purchaseArray.push(...purchases);
+      }
+      successMetadata.push(...transformedEvent.metadata);
+    } else if (!isHttpStatusSuccess(transformedEvent?.statusCode)) {
+      failureResponses.push(transformedEvent);
+    }
+  }
+  const attributeArrayChunks = _.chunk(attributesArray, 75);
+  const eventsArrayChunks = _.chunk(eventsArray, 75);
+  const purchaseArrayChunks = _.chunk(purchaseArray, 75);
+
+  const maxNumberOfRequest = Math.max(
+    attributeArrayChunks.length,
+    eventsArrayChunks.length,
+    eventsArrayChunks.length,
+  );
+  const responseArray = [];
+  for (let i = 0; i < maxNumberOfRequest; i += 1) {
+    const attributes = attributeArrayChunks[i];
+    const events = eventsArrayChunks[i];
+    const purchases = purchaseArrayChunks[i];
```

Comment:
When maxNumberOfRequest for attributes and events is greater than purchases, `purchaseArrayChunks[i]` would be `undefined` at some point, are we ok with sending `undefined` in `purchases` object to braze ? 🤔

Reply:
May be we should send `[]` to braze, but if we need to check if sending `undefined` for purchases, attributes or events is ok

Reply:
It should get handled by `removeUndefinedAndNullValues` in line 355. I will write test cases for this though

Reply:
Perfect. Please write test-cases for this and feel free to resolve the comments